controllers/core/openstackcontrolplane_controller.go

```diff
@@ -170,10 +172,81 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, nil
 	}
 
-	return r.reconcileNormal(ctx, instance, helper)
+	Log.Info("Looking up the current OpenStackVersion")
+	ctrlResult, err, version := openstack.ReconcileVersion(ctx, instance, helper)
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// wait until the version is initialized so we have images on the version.Status
+	if version.Status.Conditions.IsFalse(corev1beta1.OpenStackVersionInitialized) {
+		return ctrlResult, nil
+	}
+
+	if instance.Status.DeployedVersion == nil || version.Spec.TargetVersion == *instance.Status.DeployedVersion {
+		// green field deployment or no minor update in progress
+		ctrlResult, err := r.reconcileNormal(ctx, instance, version, helper)
+		if err != nil {
+			Log.Info("Error reconciling normal", "error", err)
+			return ctrl.Result{}, err
+		} else if (ctrlResult != ctrl.Result{}) {
+			Log.Info("Reconciling normal")
+			return ctrlResult, nil
+		}
+		// this will allow reconcileNormal to proceed in subsequent reconciles
+		instance.Status.DeployedVersion = &version.Spec.TargetVersion
+		return ctrl.Result{}, nil
+	} else {
+		if version.Status.Conditions.IsFalse(corev1beta1.OpenStackVersionMinorUpdateOVNControlplane) {
```

Comment:
I think it would be safer to say `!version.Status.Conditions.IsTrue` to properly handle when the condition is nil or Unknown